bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,608 @@
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeDeviceHandler} is responsible for handling the generic free@home device main communication
+ * and thing updates
+ *
+ * @author Andras Uhrin - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class FreeAtHomeDeviceHandler extends BaseThingHandler implements FreeAtHomeDeviceStateListener {
+
+    private static final String CHANNEL_URI = "channel-type:freeathomesystem:config";
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeDeviceHandler.class);
+    private FreeAtHomeDeviceDescription device = new FreeAtHomeDeviceDescription();
+    private FreeAtHomeChannelTypeProvider channelTypeProvider;
+    private TranslationProvider i18nProvider;
+    private LocaleProvider localeProvider;
+    private Locale locale;
+    private Bundle bundle;
+
+    private Map<ChannelUID, FreeAtHomeDatapointGroup> mapChannelUID = new HashMap<ChannelUID, FreeAtHomeDatapointGroup>();
+    private Map<String, ChannelUID> mapEventToChannelUID = new HashMap<String, ChannelUID>();
+
+    public FreeAtHomeDeviceHandler(Thing thing, FreeAtHomeChannelTypeProvider channelTypeProvider,
+            TranslationProvider i18nProvider, LocaleProvider localeProvider) {
+        super(thing);
+
+        this.channelTypeProvider = channelTypeProvider;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.locale = localeProvider.getLocale();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            FreeAtHomeBridgeHandler freeAtHomeBridge = null;
+            Map<String, String> properties = getThing().getProperties();
+            Bridge bridge = this.getBridge();
+
+            logger.debug("Start creating device - device id: {}", device.getDeviceId());
+
+            if (bridge != null) {
+                ThingHandler handler = bridge.getHandler();
+
+                if (handler instanceof FreeAtHomeBridgeHandler bridgeHandler) {
+                    freeAtHomeBridge = bridgeHandler;
+
+                    if (freeAtHomeBridge != null) {
+                        String locDeviceId = properties.get("deviceId");
```

Comment:
I did not foreseen the manual configuration because, the configuration is not known by the user because it is not written on any device. Therefore I use the property to provide an unmodifiable field for the deviceID, which is detected automatically.
The device is hidden, only the bridge can be generated manually.

Reply:
Is it correct that this property is only set by discovery? In order to support manual configuration (and thus file-based configuration) - like documented - you simply need to change this into a parameter. The discovery will be unaffected since parameters are passed as properties as well. File-based configuration is still possible by first using the discovery to create Things and then copy'n'pasting the needed parameters.

Reply:
I thought this method also. However it could be critical if the user changes the configuration in working openHAB setup.

It means the initial setup is done with device ID1 (4x  binary switch device) and change to an ID2 (room temperature controller device). Because the bridge can communicate with both but the Datapoints are completlz different and the Thing will have unexpected function and also the devices will be controlled strange via openHAB

Is it possible to define this configuration "read-only" which can be edited only by the binding itself or during the text-file based configuration?
(but no editing on the  GUI possible)

Otherwise it could be pretty dangerous due to the unexpected behaviour of the Things and free@home devices point of view.